blueprints/services/dagjson-over-http/server.go

```diff
@@ -68,31 +67,42 @@ func (x GoServerImpl) GoDef() cg.Blueprint {
 			Data: bmDecl,
 			Src: `
 		case env.{{.MethodName}} != nil:
-			ch, err := s.{{.MethodName}}({{.ContextBackground}}(), env.{{.MethodName}})
+			ch, err := s.{{.MethodName}}(request.Context(), env.{{.MethodName}})
 			if err != nil {
 				{{.LoggerVar}}.Errorf("service rejected request (%v)", err)
 				writer.Header()["Error"] = []string{err.Error()}
 				writer.WriteHeader(500)
 				return
 			}
-			for resp := range ch {
-				var env *{{.ReturnEnvelope}}
-				if resp.Err != nil {
-					env = &{{.ReturnEnvelope}}{ Error: &{{.ErrorEnvelope}}{Code: {{.EdelweissString}}(resp.Err.Error())} }
-				} else {
-					env = &{{.ReturnEnvelope}}{ {{.MethodName}}: resp.Resp }
-				}
-				var buf {{.BytesBuffer}}
-				if err = {{.IPLDEncodeStreaming}}(&buf, env, {{.DAGJSONEncode}}); err != nil {
-					{{.LoggerVar}}.Errorf("cannot encode response (%v)", err)
-					continue
-				}
-				buf.WriteByte("\n"[0])
-				writer.Write(buf.Bytes())
-				if f, ok := writer.({{.HTTPFlusher}}); ok {
-					f.Flush()
+
+			writer.WriteHeader(200)
+			if f, ok := writer.({{.HTTPFlusher}}); ok {
+				f.Flush()
+			}
```

Comment:
Why are we writing out the header and flushing here? Trying to be nice to the client, or something having to do with making context cancellation easier?

Reply:
It's mostly about synchronization during tests, the client call doesn't return the channel to the caller until the HTTP response headers are received. Before this change, the server doesn't send the headers until the first result is written and flushed, which is surprising behavior (and confusing and annoying during tests), so explicitly writing the response code and flushing the headers causes the client to always return the channel before any results are even sent.

Reply:
In a parallel PR https://github.com/ipld/edelweiss/pull/48, when a method is marked cachable it attaches an ETag to the response header. This cannot be done until after the request body is generated (below). It sounds like this optimization here won't work for that case?

Reply:
(I think you meant response body, not request body?) If we can't flush headers until the entire response body is generated, then we need to buffer the whole response and all of this becomes much easier. Is that what we want?

Reply:
Sounds good.